File: src/edu/leetcode/medium/array/MinimumSizeSubarraySum.java
package edu.leetcode.medium.array;

/**
 * Created by hungduong on 1/14/17.
 */
public class MinimumSizeSubarraySum {
    //Use Sliding window
    public static int minSubArrayLen(int target, int[] nums) {
        if(nums == null || nums.length == 0)
            return 0;

        int sum = 0;
        int minLen = Integer.MAX_VALUE;
        int left = 0;

        for(int right = 0; right < nums.length; right++) {
            sum += nums[right];

            while(sum >= target) {
                minLen = Math.min(right - left + 1, minLen);
                sum -= nums[left];
                left++;
            }
        }

        return minLen != Integer.MAX_VALUE ? minLen : 0;
    }

    public static void main(String[] args) {
        int[] nums = {2,3,1,2,4,3};
        System.out.print(minSubArrayLen(7, nums));
    }

}

File: src/edu/python/leetcode/MinimumSizeSubarraySum.py
class Solution:
    # https://leetcode.com/problems/longest-substring-without-repeating-characters/
    # Array - Sliding Window - 2 pointers
    def MinimumSizeSubarraySum(self, target: int, nums: List[int]) -> int:
        sum = 0
        min_len = math.inf
        left = 0
        for right in range(len(nums)):
            sum += nums[right]

            while sum >= target:
                min_len = min(min_len, right - left + 1)
                sum -= nums[left]
                left += 1
        
        return min_len if min_len != math.inf else 0